plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/SingletonIdentityCacheMapping.java

```diff
@@ -38,7 +38,7 @@ public int hashCode()
         @Override
         public boolean equals(Object obj)
         {
-            return obj instanceof SingletonIdentityCacheMapping;
+            return obj instanceof SingletonIdentityCacheKey;
```

Comment:
Can we have a test that would expose the problem?

Reply:
I started adding a test which puts two keys into a `HashMap` and checks that the second `put` will overwrite the first. And it turns out that `HashMap` will check equality of identities before calling `equals` (as an optimization for `null`s), so it actually works. At least with the version of JDK I have, of course.

Reply:
Ok. It would be good to have such test anyway.